# HG changeset patch
# User Pascal A. Dupuis  <Pascal.Dupuis@worldonline.be>
# Date 1282932244 14400
#      Fri Aug 27 14:04:04 2010 -0400
# Node ID 5074494f01a9376ba66028593f19affa2d7ad4b5
# Parent  2a25f282e965201b3ecfcc8e1ecb2eaeba24a0df
allow hold to work if given axis handle

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2010-08-27  Pascal A. Dupuis  <Pascal.Dupuis@worldonline.be>
+
+	* plot/hold.m: If arg is axis handle, get corresponding figure
+	handle from axis parent property.
+
 2010-08-27  Ben Abbott <bpabbott@mac.com>
 
 	* plot/__gnuplot_print__.m: Properly restore fontsize after eps output.
 	* plot/plotyy.m: Modified demo for changeset 10912 9abc67b4bd4f.
 	* plot/__fltk_print__.m, plot/__go_draw_axes__.m,
 	plot/__go_draw_figure__.m, plot/__gnuplot_print__.m,
 	plot/gnuplot_drawnow.m, plot/print.m, plot/private/__ghostscript__.m,
 	plot/private/__tight_eps_bbox__.m: Unify gnuplot printing with the
diff --git a/scripts/plot/hold.m b/scripts/plot/hold.m
--- a/scripts/plot/hold.m
+++ b/scripts/plot/hold.m
@@ -49,16 +49,17 @@
 ## @seealso{ishold, cla, newplot, clf}
 ## @end deftypefn
 
 function hold (varargin)
 
   if (nargin > 0 && numel (varargin{1}) == 1 && ishandle (varargin{1})
       && strcmp (get (varargin{1}, "type"), "axes"))
     [ax, varargin, nargs] = __plt_get_axis_arg__ ("hold", varargin{:});
+    fig = get (ax, "parent");
   elseif (nargin > 0 && numel (varargin{1}) > 1 && ishandle (varargin{1}))
     print_usage ();
   else
     ax = gca ();
     fig = gcf ();
     nargs = numel (varargin);
   endif
 
